lib/handler/redirect.js

```diff
@@ -3,18 +3,22 @@
 const { InvalidArgumentError } = require('../core/errors')
 const util = require('../core/util')
 
+const redirectableStatusCodes = [300, 301, 302, 303, 307, 308]
+
 class RedirectHandler {
-  constructor (agent, opts, handler) {
+  constructor (agent, { maxRedirections, history, ...opts }, handler) {
     this.agent = agent
     this.location = null
     this.abort = null
     this.opts = opts
+    this.maxRedirections = maxRedirections
     this.handler = handler
+    this.history = []
   }
 
-  onConnect (abort) {
+  onConnect (abort, context) {
     this.abort = abort
-    this.handler.onConnect(abort)
+    this.handler.onConnect(abort, { ...context, history: this.history })
```

Comment:
I'm thinking maybe we should put `history` as a symbol we make public as to avoid collisions in the future?

Reply:
We could, but I think the external API might look a little complicated.